src/main/java/br/com/vaasschool/security/AuthenticationService.java

```diff
@@ -0,0 +1,23 @@
+package br.com.vaasschool.security;
+
+import br.com.vaasschool.repository.UserRespository;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthenticationService implements UserDetailsService {
+
+    private final UserRespository userRespository;
+
+    public AuthenticationService(UserRespository userRespository) {
+        this.userRespository = userRespository;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String userName) throws UsernameNotFoundException {
+        return userRespository.findByEmail(userName)
+                .orElseThrow(() -> new UsernameNotFoundException("Error"));
```

Comment:
Ana achei que acabou ficando muito genérica essa mensagem....oque você acha? É que error pode indicar qualquer tipo de erro....poderia ser algo como:  user `nome do user` not found, oque acha?